neurolang/probabilistic/query_resolution.py

```diff
@@ -207,4 +269,4 @@ def _build_probabilistic_program(det_edb, pfact_db, pchoice_edb, prob_idb):
     # solution of their probability and antecedent
     prob_idb = _discard_query_based_probfacts(prob_idb)
     cpl.walk(prob_idb)
-    return cpl
+    return cpl, prob_idb
```

Comment:
Need to return the new prob IDB as it's been modified by the function (removing probabilistic encoding rules (PERs) that have been translation to one probabilistic fact table and a deterministic rule that computes the probabilistic fact table). Keeping it in the program leads to some errors later in the probabilistic query resolution

Reply:
why wasn't this an error before then?